### ISSUE TITLE
Configuring backend?

### ISSUE THREAD

**Issue by ksaric** [obsidiansystems/obelisk#222]
I'm kind of confused. Which is not odd, I'm confused most of the time...

Anyway, I thought that this is a complete web/mobile/whatever stack replacement. But I don't see a way to actually extend the backend to include the functionality I want _if I use *obelisk*_.
For example, here - https://github.com/obsidiansystems/obelisk/blob/master/lib/backend/src/Obelisk/Backend.hs#L19-L27 I can only see a trivial piece of configuration being passed it.

So the point of confusion is this - is this supposed to be a "real" backend or is this here just to serve all the required files for the frontend (using JSaddle or other options)? 
Is this a "frontend backend" or a "backend backend"? 

And how would you even use it in this state, in either of these cases?

Thanks.

**Comment by ryantrinkle** [comment]
Hi @ksaric!  Right now, the released version (i.e. master branch) is mostly focused on serving the frontend in an efficient way that behaves the same on all platforms - web, iOS, and Android.  However, the [routes branch](https://github.com/obsidiansystems/obelisk/tree/routes) is getting close to release, and it will add cross-platform routing support as well as a considerably more flexible approach to the backend configuration.  If you have a chance to take a look, I'd love to get your feedback on that branch before we release it!

**Comment by rprije** [comment]
I've upgraded to the routes version and I'm still pretty unsure how to use it. Previously I mostly ignored the Backend and replaced it with something I wrote myself. But routes has made this harder and also demands that I return a Snap. I'm also not really sure what I'm supposed to do with the "R backendroute" object my function receives.

Up until this change and through sidestepping the provided backend infrastructure, I was able to use servant-server. I tried getting servant-snap working to glue it all back together, but that library has fallen a bit behind and won't compile without downgrading other libraries.

In summary
* it'd be nice if this wasn't so snap-centric. Though I understand obelisk is intended to be an opinionated platform, I feel it probably doesn't need to be quite so opinionated about the backend
* some documentation explaining how to build a working "Backend" type would be helpful.

**Comment by rprije** [comment]
Quick update. I was able to work around my problem. I just threw away the serve function my _backend_run function receives and stopped thinking about how to integrate Snap into my code. So functionality doesn't seem any more limited for me than before the routes change.

I still don't know how the new route stuff is intended to be used, but that no longer seems necessary to understand in order to use the platform for what I've been using it for.
